### PR TITLE
Item 6834: Display Source Types: Add category field to exp.dataclasses

### PR DESCRIPTION
**Pull request by XingY**

### CODE REVIEW

File: experiment/resources/schemas/dbscripts/postgresql/exp-20.000-20.001.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE exp.DataClass ADD COLUMN Category VARCHAR(500) NULL;
```
Comment (labkey-susanh) [inline experiment/resources/schemas/dbscripts/postgresql/exp-20.000-20.001.sql]: 500 seems quite large for the category.  I expect the strings to be no more than 20 characters in general.

Comment (labkey-susanh) [reply experiment/resources/schemas/dbscripts/postgresql/exp-20.000-20.001.sql]: I also wonder if we don't want Category to be an EnumTableType.

File: experiment/src/org/labkey/experiment/insertDataClass.jsp [hunk 2]
```diff
@@ -94,6 +94,8 @@
         </div>
     </div>
 
+    <%-- TODO: dataclass.category field not yet supported for LKS insert --%>
```
Comment (labkey-susanh) [inline experiment/src/org/labkey/experiment/insertDataClass.jsp:97]: @cnathe For your current story on creating source types, you are also making changes to the LKS view of this, right?  It seems like we should add a dropdown for specifying a category, in which case using an Enum is even more appropriate.

File: query/src/org/labkey/query/QueryServiceImpl.java [hunk 3]
```diff
@@ -2740,6 +2742,19 @@ public void registerSchemaLinkAction(@NotNull Class<? extends Controller> action
...
     }
 
+    @Override
+    public void registerQueryIconURLProvider(QueryIconURLProvider queryIconProvider)
+    {
+        _queryIconURLProviders.add(queryIconProvider);
+    }
+
+    @Override
+    @NotNull public List<QueryIconURLProvider> getQueryIconURLProvider()
```
Comment (labkey-susanh) [inline query/src/org/labkey/query/QueryServiceImpl.java]:
```suggestion
    @NotNull public List<QueryIconURLProvider> getQueryIconURLProviders()
```
Returning a list, so plural seems appropriate

**Review by labkey-susanh — CHANGES_REQUESTED** [review]

File: experiment/src/org/labkey/experiment/insertDataClass.jsp (hunk 2 above)
Comment (cnathe) [reply experiment/src/org/labkey/experiment/insertDataClass.jsp:97]: Yes, we can add this part as a task to the Data Class Designer story that I am working on. I will do that now so I don't forget.

**Review by labkey-susanh — APPROVED** [review]

**Pull request merged**